### DISCUSSION TITLE
[Question] Is it possible to create dependent/ chained filter fields?

### DISCUSSION THREAD

**Discussion by vitormhenrique** [carltongibson/django-filter#1431] [Q&A]
sorry pressed enter without adding description

I want to create a filter with where the fields should filter themselves, for example a list of cities on one filed tat will be filtered by the state selected.

I looked the previous issues and can't figure it out how to implement said filter....

**Comment by urzbs** [comment]
I have the exact same question and cant figure it out aswell. 
It would make much sense to have this feature, since it enhances the user experience by a lot - when filter options on foreign key fields that would furthermore yield no results are excluded from the choices.

For sure it would require some ajax to the backend when a filter has been selected that influences the other fields. Alternativly I tried to manipulate the html with javascript - but that didnt work out very well.

Some coverage would be very appreciated, maybe more people have this problem too.

**Comment by vitormhenrique** [comment]
I found another djang app that could do that, but forgot the name now, but ended up opted to just use javascript to make requests and update the filter. 
There is a simple tutorial [here](https://simpleisbetterthancomplex.com/tutorial/2018/01/29/how-to-implement-dependent-or-chained-dropdown-list-with-django.html) if you wanna take a look.

**Comment by GerardoGa** [comment]
You can create dependent fields by using a [DjangoSelect2](https://github.com/codingjoe/django-select2) widget and specifying the dependent fields inside the form.

```
class CompanyFilterSet(django_filters.FilterSet):
    address_province = django_filters.ModelChoiceFilter(queryset=<Province Queryset>, widget=ProvinceWidget()
    address_city = django_filters.ModelChoiceFilter(queryset=<City Queryset>, widget=CityWidget(dependent_fields={'address_province': 'province'}))
   
    class Meta:
        model = Company
        fields = []
 
```

**Comment by irvan-k** [comment]
can you tell me more about Django select2 I have work with that I got confused isn't there anyway ?
@GerardoGa

**Comment by GerardoGa** [comment]
Django Select2 is a very useful library as it uses Select2 to render a select inside a form. It enhances the posibilities of a select, for exemple, by filtering results depending on another select value. Specifying  **dependent_fields={'address_province': 'province'}** the queryset of the field is filtered by the address_province selected value, excluding the other results. 

Another advantage of DjangoSelect2 is that it paginates the options of a select reducing form rendering time. It also adds a browser to search inside the options.

You can find more information [here](https://django-select2.readthedocs.io/en/latest/).

**Comment by irvan-k** [comment]
can you show me code i cant understand django select @GerardoGa
